src/Servers/HttpSys/test/FunctionalTests/ResponseTrailersTests.cs

```diff
@@ -0,0 +1,384 @@
+                Assert.Equal(HttpVersion.Version20, response.Version);
+                Assert.Empty(response.TrailingHeaders);
+            }
+        }
+
+        // https://tools.ietf.org/html/rfc7230#section-4.1.2
+        private readonly List<string> _disallowedTrailers = new List<string>()
+        {
+            // Message framing headers.
+            HeaderNames.TransferEncoding, HeaderNames.ContentLength,
+
+            // Routing headers.
+            HeaderNames.Host,
+
+            // Request modifiers: controls and conditionals.
+            // rfc7231#section-5.1: Controls.
+            HeaderNames.CacheControl, HeaderNames.Expect, HeaderNames.MaxForwards, HeaderNames.Pragma, HeaderNames.Range, HeaderNames.TE,
+
+            // rfc7231#section-5.2: Conditionals.
+            HeaderNames.IfMatch, HeaderNames.IfNoneMatch, HeaderNames.IfModifiedSince, HeaderNames.IfUnmodifiedSince, HeaderNames.IfRange,
+
+            // Authentication headers.
+            HeaderNames.WWWAuthenticate, HeaderNames.Authorization, HeaderNames.ProxyAuthenticate, HeaderNames.ProxyAuthorization, HeaderNames.SetCookie, HeaderNames.Cookie,
+
+            // Response control data.
+            // rfc7231#section-7.1: Control Data.
+            HeaderNames.Age, HeaderNames.Expires, HeaderNames.Date, HeaderNames.Location, HeaderNames.RetryAfter, HeaderNames.Vary, HeaderNames.Warning,
+
+            // Content-Encoding, Content-Type, Content-Range, and Trailer itself.
+            HeaderNames.ContentEncoding, HeaderNames.ContentType, HeaderNames.ContentRange, HeaderNames.Trailer
+        };
+
+        [ConditionalFact]
+        [MinimumOSVersion(OperatingSystems.Windows, "10.0.19505", SkipReason = "Requires HTTP/2 Trailers support.")]
+        public async Task ResponseTrailers_ProhibitedTrailers_Blocked()
+        {
+            using (Utilities.CreateDynamicHttpsServer(out var address, httpContext =>
+            {
+                Assert.True(httpContext.Response.SupportsTrailers());
+                foreach (var header in _disallowedTrailers)
+                {
+                    Assert.Throws<InvalidOperationException>(() => httpContext.Response.AppendTrailer(header, "value"));
+                }
+                return Task.FromResult(0);
+            }))
+            {
+                var response = await SendRequestAsync(address);
+                response.EnsureSuccessStatusCode();
+                Assert.Equal(HttpVersion.Version20, response.Version);
+                Assert.Empty(response.TrailingHeaders);
+            }
+        }
+
+        [ConditionalFact]
+        [MinimumOSVersion(OperatingSystems.Windows, "10.0.19505", SkipReason = "Requires HTTP/2 Trailers support.")]
+        public async Task ResponseTrailers_NoBody_TrailersSent()
+        {
+            using (Utilities.CreateDynamicHttpsServer(out var address, httpContext =>
+            {
+                httpContext.Response.DeclareTrailer("trailername");
```

Comment:
Is this important to the test?

Reply:
It's not the most critical test here bit it's good to know that it works as expected.